src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/SetPiFromGauge.hpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <random>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "Domain/Tags.hpp"
+#include "Domain/TagsTimeDependent.hpp"
+#include "Evolution/Initialization/Tags.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Tags/GaugeCondition.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace GeneralizedHarmonic::gauges {
+/*!
+ * \brief Set \f$\Pi_{ab}\f$ from the gauge source function.
+ *
+ * This is necessary to ensure the initial data is in the desired evolution
+ * gauge.
+ */
+template <size_t Dim>
+struct SetPiFromGauge {
+ public:
+  using return_tags =
+      tmpl::list<GeneralizedHarmonic::Tags::Pi<Dim, Frame::Inertial>>;
+  using argument_tags =
+      tmpl::list<::Tags::Time, domain::Tags::Mesh<Dim>,
```

Comment:
Follow-up to other comment, if `Tags::Time` is invalid, should this be `Tags::InitialTime`?